routes/movies.test.js

```diff
@@ -1,39 +1,233 @@
+  //   });
+  //   it("should return an error message if movie fails to be created", async () => {
+  //     expect(false).toEqual(true);
+  //     // expect status code == 400
+  //   });
+  // });
+
+  // describe("PUT /:id", () =>{
+  //   it("should return the updated movie on success", async () => {
+  //     expect(false).toEqual(true);
+  //   });
+  //   it("should return an error message if movie fails to be updated", async () => {
+  //     expect(false).toEqual(true);
+  //   });
+  // });
+
+  // describe("DELETE /:id", () =>{
+  //   it("should return a message on success", async () => {
+  //     expect(false).toEqual(true);
+  //   });
+  //   it("should return a error message if movie fails to be deleted", async () => {
+  //     expect(false).toEqual(true);
+  //   });
+  // });
+
+  //   describe("GET /movies/genres/:genreName", () =>{
+  //   it("should return an array of movies on success", async () => {
+  //     // TODO: Mock the correct data interface method
+  //     const res = await request(server).get("/movies/genres/Short");
+
+  //     expect(res.statusCode).toEqual(200);
+  //     expect(Array.isArray(res.body)).toEqual(true);
+  //     expect(res.body.error).not.toBeDefined();
+  //   });
+  //   it("should return an empty array if no movies match genre", async () => {
+  //     // TODO: Mock the correct data interface method
+  //     const res = await request(server).get("/movies/genres/UEOA921DI");
+
+  //     expect(res.statusCode).toEqual(200);
+  //     expect(res.body.length).toEqual(0);
+  //     expect(res.body.error).not.toBeDefined();
+  //   });
+  //   it("should return an error message on error", async () => {
+  //     // TODO: Mock the correct data interface method
+
+  //     const res = await request(server).get("/movies/genres/Short");
+
+  //     expect(res.statusCode).toEqual(500);
+  //     expect(res.body.error).toBeDefined();
+  //   });
+  // });
+
+  describe("test get all comments for a movie GET /movies/:id/comments", () => {
+    it("should return an array of comments on success", async () => {
+      movieData.getAllComments.mockResolvedValue([{
+        _id: "5a9427648b0beebeb6957d04",
+        name: "Lori Blankenship",
+        email: "lori_blankenship@fakegmail.com",
+        movie_id: "573a1391f29313caabcd9688",
+        text: "Totam delectus ullam exercitationem maiores. Praesentium velit veniam ...",
+        date: "1974-08-06T06:47:37.000+00:00"
+      }]);
+
+      const res = await request(server).get("/movies/573a1391f29313caabcd9688/comments");
+
+      expect(res.statusCode).toEqual(200);
+      expect(Array.isArray(res.body)).toEqual(true);
+      expect(res.body.error).not.toBeDefined();
+
+    });
+    it("should return an error message on error", async () => {
+      movieData.getAllComments.mockResolvedValue({});
+
+      const res = await request(server).get("/movies");
+
+      expect(res.statusCode).toEqual(500);
+      expect(res.body.error).toBeDefined();
```

Comment:
The route handler under test looks like this: 
```
router.get("/:id/comments", async(req, res) => {
  const result = await movieData.getAllComments(req.params.id)
  if (result) {
    res.status(200).send(result);
  } else {
    resultStatus = 404;
  }
})
```
The two responses from `movieData.getAllComments` that it knows how to handle are something and nothing (null, undefined). The mocked return value here on line 121 is something, so I would expect the route handler to return a status code of 200 and a body that doesn't include an error.